[2025-02-09T03:46:31.072+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-02-09T03:46:31.105+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_MusicStreamPipeline.MusicStream_transformation_song manual__2025-02-09T03:46:27.428282+00:00 [queued]>
[2025-02-09T03:46:31.274+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_MusicStreamPipeline.MusicStream_transformation_song manual__2025-02-09T03:46:27.428282+00:00 [queued]>
[2025-02-09T03:46:31.274+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 1
[2025-02-09T03:46:31.294+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): MusicStream_transformation_song> on 2025-02-09 03:46:27.428282+00:00
[2025-02-09T03:46:31.304+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_MusicStreamPipeline', 'MusicStream_transformation_song', 'manual__2025-02-09T03:46:27.428282+00:00', '--job-id', '373', '--raw', '--subdir', 'DAGS_FOLDER/etl_dag.py', '--cfg-path', '/tmp/tmpmku5xm1h']
[2025-02-09T03:46:31.307+0000] {standard_task_runner.py:91} INFO - Job 373: Subtask MusicStream_transformation_song
[2025-02-09T03:46:31.309+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=335) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-02-09T03:46:31.310+0000] {standard_task_runner.py:63} INFO - Started process 340 to run task
[2025-02-09T03:46:31.353+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_MusicStreamPipeline.MusicStream_transformation_song manual__2025-02-09T03:46:27.428282+00:00 [running]> on host e5529003cf96
[2025-02-09T03:46:31.446+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Le Huynh Thanh Duong' AIRFLOW_CTX_DAG_ID='etl_MusicStreamPipeline' AIRFLOW_CTX_TASK_ID='MusicStream_transformation_song' AIRFLOW_CTX_EXECUTION_DATE='2025-02-09T03:46:27.428282+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-02-09T03:46:27.428282+00:00'
[2025-02-09T03:46:31.447+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-02-09T03:46:31.505+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-02-09T03:46:31.506+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etls/transform.py", line 7, in process_song
    stg_artist = song_data[['artist_id', 'artist_name', 'artist_location', 'artist_latitude', 'artist_longitude']]
                 ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/frame.py", line 3899, in __getitem__
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/indexes/base.py", line 6115, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/indexes/base.py", line 6176, in _raise_if_missing
    raise KeyError(f"None of [{key}] are in the [{axis_name}]")
KeyError: "None of [Index(['artist_id', 'artist_name', 'artist_location', 'artist_latitude',\n       'artist_longitude'],\n      dtype='object')] are in the [columns]"
[2025-02-09T03:46:31.519+0000] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=etl_MusicStreamPipeline, task_id=MusicStream_transformation_song, execution_date=20250209T034627, start_date=20250209T034631, end_date=20250209T034631
[2025-02-09T03:46:31.529+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 373 for task MusicStream_transformation_song ("None of [Index(['artist_id', 'artist_name', 'artist_location', 'artist_latitude',\n       'artist_longitude'],\n      dtype='object')] are in the [columns]"; 340)
[2025-02-09T03:46:31.564+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-02-09T03:46:31.581+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-09T03:46:31.584+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
